android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/activity/ActivitySchemaCache.java

```diff
@@ -0,0 +1,156 @@
+package org.sagebionetworks.bridge.android.manager.activity;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.collect.Maps;
+
+import org.sagebionetworks.bridge.rest.model.Activity;
+import org.sagebionetworks.bridge.rest.model.ActivityType;
+import org.sagebionetworks.bridge.rest.model.CompoundActivity;
+import org.sagebionetworks.bridge.rest.model.SchemaReference;
+import org.sagebionetworks.bridge.rest.model.SurveyReference;
+import org.sagebionetworks.bridge.rest.model.TaskReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * A cache of mappings from activities' identifiers (tasks, surveys, compound activities) to
+ * (SchemaReference, SurveyReference). This is to help identify how an activity's return data should
+ * be formatted.
+ */
+public class ActivitySchemaCache {
+    private static final Logger LOG = LoggerFactory.getLogger(ActivitySchemaCache.class);
+    @NonNull
+    private final Map<String, SchemaReference> taskIdentifierToSchema;
```

Comment:
I think that this data structure could be better.  5 maps that are coupled and need to be maintained together but are actually separate variables, will make it easy for someone who is not familiar with their inter-dependence to create a bug.  

I would suggest re-evaluating how this could be stored as a single Map.  My first thought would be to have a Map<String, TypeValueActivity>, where TypeValueActivity would hold the activity's type (task, survey, compound) and a getter for the Value which does a switch on the Type and casts the Object value as either SurveyReference, SchemaReference, or CompoundActivity.